VirtoCommerce.MarketingModule.Web/Scripts/promotion/blades/catalog-entry-list.js

```diff
@@ -0,0 +1,139 @@
+                                    entry.imgSrc = entry.imageUrl;
+                                }
+                                return entry;
+                            });
+                            bladeNavigationService.closeBlade(catalogBlade);
+                        },
+                        canExecuteMethod: function () {
+                            return true;
+                        }
+                    }]
+                };
+
+                catalogBlade.options = {
+                    selectedItemIds: blade.productIds,
+                    showCheckingMultiple: false,
+                    checkItemFn: function (listItem, isSelected) {
+                        if (listItem.type == 'category') {
+                            blade.error = 'Must select Product';
+                            listItem.selected = undefined;
+                        } else {
+                            if (isSelected) {
+                                if (_.all(blade.selectedListEntries, function (x) { return x.id != listItem.id; })) {
+                                    blade.selectedListEntries.push(listItem);
+                                }
+                            }
+                            else {
+                                blade.selectedListEntries = _.reject(blade.selectedListEntries, function (x) { return x.id == listItem.id; });
+                            }
+                            blade.error = undefined;
+                        }
+                    }
+                };
+
+                bladeNavigationService.showBlade(catalogBlade, blade);
+            }
+
+            $scope.blade.toolbarCommands = [
+                {
+                    name: "platform.commands.refresh", icon: 'fa fa-refresh',
+                    executeMethod: function (blade) {
+                        blade.refresh();
+                    },
+                    canExecuteMethod: function () { return true; }
+                },
+                {
+                    name: "platform.commands.add", icon: 'fa fa-plus',
+                    executeMethod: function () {
+                        openCatalogBlade();
+                    },
+                    canExecuteMethod: function () {
+                        return true;
+                    }
+                },
+                {
+                    name: "platform.commands.delete", icon: 'fa fa-trash-o',
+                    executeMethod: function () {
+                        var grid = blade.$scope.gridApi.grid;
+                        var selectedRows = grid.api.selection.getSelectedRows();
+                        for (var entry of selectedRows) {
+                            grid.options.data = _.filter(grid.options.data, function (item) {
+                                return item.id != entry.id;
+                            });
+                        }
+
+                        blade.selectedListEntries = grid.options.data.slice();
```

Comment:
Do we need copy here?

Reply:
This copy is needed for catalog items blade which could change it.